azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/PackageMojo.java

```diff
@@ -273,12 +274,19 @@ protected FunctionCoreToolsHandler getFunctionCoreToolsHandler(final CommandHand
         return new FunctionCoreToolsHandlerImpl(this, commandHandler);
     }
 
-    protected void installExtension(final FunctionCoreToolsHandler handler) throws Exception {
+    protected void installExtension(final FunctionCoreToolsHandler handler,
+                                    Map<String, FunctionConfiguration> configMap) throws Exception {
         info("");
         info(INSTALL_EXTENSIONS);
-        handler.installExtension();
+        handler.installExtension(getFunctionBindingClasses(configMap));
```

Comment:
Instead of passing the `configMap` deeply. I prefer to have a util method to check if installing the extension is needed before trigger the installing command